blockchain.go

```diff
@@ -193,17 +193,19 @@ var defaultConfig = func() config {
 // Option is a function applying a change to the emulator config.
 type Option func(*config)
 
-// WithServicePublicKey sets the service key from a public key.
-func WithServicePublicKey(
```

Comment:
I have doubts if this API should be removed. I think it would be better if `WithServicePrivateKey` would be added, and that one shouldn't have `servicePublicKey` property as it can be inferred from the private key. This is especially true or cases where you don't need these contracts so the private key is not necessary.

Reply:
great. I put this back. it turns out that i don't need to do this -- in fact, in regular usage, this function shouldn't be used at all (since we do want to keep the default service private key :)

Reply:
Exactly, that was worrisome to me.